wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+        if self.does_project_exist(name=name):
+            self.o_api.delete_project(name=name)
+            if not self.does_project_exist(name=name, wait=wait):
+                raise TimedOutError('project {n} was not removed within {w} sec'.format(n=name,
+                                                                                        w=wait))
+
+    def scale_entity(self, namespace, name, replicas, wait=60):
+        """Allows to scale up/down entities.
+        One of cases when this is necessary is emulation of stopping/starting appliance
+
+        Args:
+            namespace: openshift namespace
+            name: entity name. it can be either stateless Pod from DeploymentConfig or StatefulSet
+            replicas: number of replicas 0..N
+            wait: time to wait for scale up/down
+        Return: None
+        """
+        # only dc and statefulsets can be scaled
+        st_api = self.kclient.AppsV1beta1Api()
+
+        scale_val = self.kclient.V1Scale(spec=self.kclient.V1ScaleSpec(replicas=replicas))
+        if self.is_deployment_config(name=name, namespace=namespace):
+            self.o_api.patch_namespaced_deployment_config_scale(name=name, namespace=namespace,
+                                                                body=scale_val)
+
+            def check_scale_value():
+                got_scale = self.o_api.read_namespaced_deployment_config_scale(name=name,
+                                                                               namespace=namespace)
+                return 0 if got_scale.spec.replicas is None else int(got_scale.spec.replicas)
+
+        elif self.is_stateful_set(name=name, namespace=namespace):
+            # replace this code with stateful_set_scale when kubernetes shipped with openshift
+            # client gets upgraded
+            st_spec = self.kclient.V1beta1StatefulSetSpec
+            st = self.kclient.V1beta1StatefulSet(spec=st_spec(replicas=replicas))
+            st_api.patch_namespaced_stateful_set(name=name, namespace=namespace, body=st)
+
+            def check_scale_value():
+                got_scale = st_api.read_namespaced_stateful_set(name=name, namespace=namespace)
+                return 0 if got_scale.status.replicas is None else int(got_scale.spec.replicas)
+        else:
+            raise ValueError("This name is not found among deployment configs or stateful sets")
+        self.logger.info("scaling entity %s to %s replicas", name, replicas)
+        wait_for(check_scale_value, num_sec=wait, fail_condition=lambda val: val != replicas)
+
+    def get_project_by_name(self, project_name):
+        """Returns only the selected Project object"""
+        return Project(self, project_name)
+
+    def is_vm_running(self, vm_name):
+        """Emulates check is vm(appliance) up and running
+
+        Args:
+            vm_name: project(namespace) name
+        Return: True/False
+        """
+        logger.info("checking all pod statuses")
```

Comment:
**REQUIRED**: `self.` and printout of vm_name as well, better debugging

Reply:
In general, check all the logger calls to have `self.` and have classic-style formatters, not `.format` style formatters.